# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [1]:
## Code goes here
import re
import csv 

dirtyrecweek = dict()

with open('recweekdata.txt') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    i = 0
    for row in csv_reader:
        if line_count == 0:
            print(f'Column names are {", ".join(row)}') 
        else:
            dirtyrecweek[i]={"firstname":row[0],"lastname":row[1], "birthmonth":row[2], "birthday":row[3], "birthyear":row[4],"course":row[5],"phonenumber":row[6] }
        i += 1
        line_count += 1

dirtyrecweek

Column names are firstname, lastname, birthmonth, birthday, birthyear, course, phonenumber


{1: {'firstname': 'name1',
  'lastname': 'lastname1',
  'birthmonth': '11',
  'birthday': '26',
  'birthyear': '1998',
  'course': 'BS LM',
  'phonenumber': '+63 918 302 6210'},
 2: {'firstname': 'name2',
  'lastname': 'lastname2',
  'birthmonth': 'May',
  'birthday': '26',
  'birthyear': '2001',
  'course': 'Management',
  'phonenumber': '927 212 6276'},
 3: {'firstname': 'name3',
  'lastname': 'lastname3',
  'birthmonth': '5',
  'birthday': '16',
  'birthyear': '1997',
  'course': 'BS CTM',
  'phonenumber': '+63 918 253 7017'},
 4: {'firstname': 'name4',
  'lastname': 'lastname4',
  'birthmonth': 'Dec',
  'birthday': '9',
  'birthyear': '97',
  'course': 'BS ITE',
  'phonenumber': '918 714 4360'},
 5: {'firstname': 'name5',
  'lastname': 'lastname5',
  'birthmonth': 'Sep',
  'birthday': '5',
  'birthyear': '97',
  'course': 'BS CTM',
  'phonenumber': '+63 927 462 1674'},
 6: {'firstname': 'name6',
  'lastname': 'lastname6',
  'birthmonth': 'March',
  'birthday': '2',
  'birthyear': '

## Birth Month Tally

In [2]:
birthmlist= []
for i in range(1,3001):
    birthmlist.append(dirtyrecweek[i]["birthmonth"]) #making a list of the birth months
    
birthmlist

['11',
 'May',
 '5',
 'Dec',
 'Sep',
 'March',
 '11',
 'Aug',
 'July',
 '9',
 'December',
 '5',
 'January',
 '1',
 '6',
 '4',
 '1',
 'Apr',
 '12',
 '3',
 '7',
 '9',
 '4',
 'Oct',
 '5',
 '12',
 'March',
 '11',
 '10',
 'May',
 '4',
 'Jul',
 '12',
 'Oct',
 '7',
 'December',
 '2',
 '8',
 'May',
 '2',
 'Nov',
 'September',
 'Jan',
 'February',
 'January',
 'Dec',
 'August',
 'Apr',
 '10',
 'March',
 '8',
 '3',
 'Dec',
 '7',
 'May',
 '9',
 'May',
 'April',
 'April',
 'August',
 '9',
 'Dec',
 'Feb',
 'November',
 'Mar',
 'September',
 'Dec',
 'November',
 'November',
 'May',
 '8',
 'December',
 'Jun',
 'Oct',
 'Sep',
 '2',
 'May',
 '9',
 'December',
 '8',
 'Feb',
 '11',
 'Jun',
 'February',
 'Jan',
 'Nov',
 'August',
 'January',
 '3',
 'Mar',
 'Aug',
 'Dec',
 'Jun',
 'November',
 '9',
 'Mar',
 'Jan',
 'Jan',
 'May',
 '5',
 'April',
 'August',
 'Dec',
 'July',
 'Mar',
 '8',
 'April',
 'May',
 '8',
 'January',
 '12',
 'May',
 'May',
 'August',
 'March',
 'Jul',
 '1',
 'Feb',
 'Dec',
 '2',
 '3',

In [9]:
uniquebirthm = set(birthmlist)

uniquebirthmdict = {months:None for months in uniquebirthm}

for i in uniquebirthmdict:
    monthtitle = input(i+": "+"Enter name: ") 
    uniquebirthmdict[i]=monthtitle

uniquebirthmdict

September: Enter name: September
Nov: Enter name: November
5: Enter name: May
2: Enter name: February
Jul: Enter name: July
Feb: Enter name: February
February: Enter name: February
July: Enter name: July
January: Enter name: January
April: Enter name: April
7: Enter name: July
March: Enter name: March
Mar: Enter name: March
11: Enter name: November
June: Enter name: June
6: Enter name: June
9: Enter name: September
May: Enter name: May
Apr: Enter name: April
Sep: Enter name: September
Jan: Enter name: January
12: Enter name: December
August: Enter name: August
1: Enter name: January
3: Enter name: March
November: Enter name: November
Dec: Enter name: December
Oct: Enter name: October
4: Enter name: April
Aug: Enter name: August
Jun: Enter name: June
8: Enter name: August
10: Enter name: October
December: Enter name: December


{'September': 'September',
 'Nov': 'November',
 '5': 'May',
 '2': 'February',
 'Jul': 'July',
 'Feb': 'February',
 'February': 'February',
 'July': 'July',
 'January': 'January',
 'April': 'April',
 '7': 'July',
 'March': 'March',
 'Mar': 'March',
 '11': 'November',
 'June': 'June',
 '6': 'June',
 '9': 'September',
 'May': 'May',
 'Apr': 'April',
 'Sep': 'September',
 'Jan': 'January',
 '12': 'December',
 'August': 'August',
 '1': 'January',
 '3': 'March',
 'November': 'November',
 'Dec': 'December',
 'Oct': 'October',
 '4': 'April',
 'Aug': 'August',
 'Jun': 'June',
 '8': 'August',
 '10': 'October',
 'December': 'December'}

In [10]:
cleanbirthmlist = [re.sub(month,uniquebirthmdict[month],month) for month in birthmlist] #clean list
cleanbirthmlist

['November',
 'May',
 'May',
 'December',
 'September',
 'March',
 'November',
 'August',
 'July',
 'September',
 'December',
 'May',
 'January',
 'January',
 'June',
 'April',
 'January',
 'April',
 'December',
 'March',
 'July',
 'September',
 'April',
 'October',
 'May',
 'December',
 'March',
 'November',
 'October',
 'May',
 'April',
 'July',
 'December',
 'October',
 'July',
 'December',
 'February',
 'August',
 'May',
 'February',
 'November',
 'September',
 'January',
 'February',
 'January',
 'December',
 'August',
 'April',
 'October',
 'March',
 'August',
 'March',
 'December',
 'July',
 'May',
 'September',
 'May',
 'April',
 'April',
 'August',
 'September',
 'December',
 'February',
 'November',
 'March',
 'September',
 'December',
 'November',
 'November',
 'May',
 'August',
 'December',
 'June',
 'October',
 'September',
 'February',
 'May',
 'September',
 'December',
 'August',
 'February',
 'November',
 'June',
 'February',
 'January',
 'November',
 'August',
 'Januar

In [11]:
cleanbirthmtally={}

for i in cleanbirthmlist:
    cleanbirthmtally.setdefault(i, cleanbirthmlist.count(i))

cleanbirthmtally

{'November': 277,
 'May': 273,
 'December': 276,
 'September': 232,
 'March': 246,
 'August': 241,
 'July': 249,
 'January': 270,
 'June': 259,
 'April': 267,
 'October': 178,
 'February': 232}

In [12]:
sortedmonthlist = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
sortedmonthtally ={}

for i in sortedmonthlist:
    sortedmonthtally[i]= cleanbirthmtally[i]
    
sortedmonthtally

{'January': 270,
 'February': 232,
 'March': 246,
 'April': 267,
 'May': 273,
 'June': 259,
 'July': 249,
 'August': 241,
 'September': 232,
 'October': 178,
 'November': 277,
 'December': 276}

In [13]:
#Checking to make sure all 3000 data was included
checkermonth = 0
for i in sortedmonthlist:
    checkermonth += sortedmonthtally[i]
    
checkermonth

3000

## Birth Day Tally

In [42]:
birthdlist= []

for i in range(1,3001):
    birthdlist.append(dirtyrecweek[i]["birthday"])
    
birthdlist

['26',
 '26',
 '16',
 '9',
 '5',
 '2',
 '10',
 '28',
 '10',
 '14',
 '8',
 '4',
 '28',
 '11',
 '6',
 '20',
 '3',
 '8',
 '28',
 '10',
 '13',
 '15',
 '15',
 '1',
 '6',
 '5',
 '4',
 '10',
 '10',
 '28',
 '6',
 '16',
 '17',
 '4',
 '14',
 '11',
 '2',
 '25',
 '3',
 '27',
 '16',
 '24',
 '13',
 '25',
 '27',
 '24',
 '23',
 '13',
 '11',
 '25',
 '22',
 '21',
 '15',
 '10',
 '8',
 '9',
 '9',
 '6',
 '24',
 '26',
 '8',
 '7',
 '22',
 '13',
 '23',
 '24',
 '9',
 '4',
 '19',
 '12',
 '19',
 '1',
 '21',
 '25',
 '17',
 '2',
 '19',
 '15',
 '23',
 '10',
 '1',
 '22',
 '18',
 '28',
 '10',
 '21',
 '15',
 '26',
 '26',
 '15',
 '16',
 '8',
 '14',
 '24',
 '6',
 '16',
 '19',
 '10',
 '12',
 '10',
 '2',
 '14',
 '16',
 '16',
 '5',
 '25',
 '21',
 '14',
 '15',
 '8',
 '22',
 '19',
 '27',
 '22',
 '26',
 '19',
 '11',
 '4',
 '27',
 '3',
 '5',
 '1',
 '23',
 '2',
 '24',
 '4',
 '3',
 '20',
 '25',
 '18',
 '23',
 '10',
 '18',
 '7',
 '3',
 '5',
 '5',
 '5',
 '8',
 '2',
 '8',
 '3',
 '15',
 '14',
 '9',
 '11',
 '6',
 '13',
 '18',
 '5',
 

In [43]:
cleanbirthdtally={}

for i in birthdlist:
    cleanbirthdtally.setdefault(i, birthdlist.count(i))

cleanbirthdtally

{'26': 116,
 '16': 115,
 '9': 98,
 '5': 100,
 '2': 106,
 '10': 111,
 '28': 104,
 '14': 102,
 '8': 111,
 '4': 110,
 '11': 105,
 '6': 99,
 '20': 91,
 '3': 90,
 '13': 119,
 '15': 121,
 '1': 112,
 '17': 116,
 '25': 116,
 '27': 102,
 '24': 108,
 '23': 116,
 '22': 103,
 '21': 123,
 '7': 88,
 '19': 113,
 '12': 98,
 '18': 107}

In [44]:
sorteddaylist = [i for i in range(1,29)] #checked the cleanbirthdtally and found that only days 1-28 were used 
sorteddaylist.sort()
sorteddaylist #made a list of arranged numbers

[1,
 2,
 3,
 4,
 5,
 6,
 7,
 8,
 9,
 10,
 11,
 12,
 13,
 14,
 15,
 16,
 17,
 18,
 19,
 20,
 21,
 22,
 23,
 24,
 25,
 26,
 27,
 28]

In [45]:
sorteddaytally ={}

for i in sorteddaylist:
    sorteddaytally[i]= cleanbirthdtally[str(i)] #assigned the values to the days of the month
   
sorteddaytally

{1: 112,
 2: 106,
 3: 90,
 4: 110,
 5: 100,
 6: 99,
 7: 88,
 8: 111,
 9: 98,
 10: 111,
 11: 105,
 12: 98,
 13: 119,
 14: 102,
 15: 121,
 16: 115,
 17: 116,
 18: 107,
 19: 113,
 20: 91,
 21: 123,
 22: 103,
 23: 116,
 24: 108,
 25: 116,
 26: 116,
 27: 102,
 28: 104}

In [18]:
#Checking to make sure all 3000 data was included
checkerday = 0
for i in range(1,29):
    checkerday += sorteddaytally[i]  
    
checkerday

3000

## Birth Year Tally

In [19]:
birthylist= []

for i in range(1,3001):
    birthylist.append(dirtyrecweek[i]["birthyear"])
    
birthylist

['1998',
 '2001',
 '1997',
 '97',
 '97',
 '1999',
 '98',
 '99',
 '97',
 '2001',
 '98',
 '97',
 '1999',
 '1999',
 '98',
 '99',
 '97',
 '1997',
 '1997',
 '97',
 '99',
 '1999',
 '97',
 '97',
 '97',
 '1999',
 '1997',
 '2000',
 '1999',
 '1999',
 '99',
 '99',
 '99',
 '1999',
 '2001',
 '1998',
 '2000',
 '2001',
 '1997',
 '1998',
 '99',
 '98',
 '1999',
 '98',
 '97',
 '98',
 '1999',
 '99',
 '97',
 '2001',
 '98',
 '1999',
 '2000',
 '98',
 '1998',
 '1998',
 '2001',
 '97',
 '98',
 '99',
 '98',
 '2000',
 '2000',
 '97',
 '97',
 '1997',
 '1997',
 '2001',
 '1998',
 '98',
 '99',
 '1999',
 '1998',
 '1997',
 '1998',
 '1997',
 '2000',
 '2000',
 '2000',
 '99',
 '97',
 '1997',
 '2000',
 '1999',
 '1999',
 '98',
 '1998',
 '97',
 '1998',
 '99',
 '97',
 '97',
 '1998',
 '97',
 '2001',
 '1999',
 '1997',
 '98',
 '98',
 '99',
 '97',
 '2001',
 '2000',
 '2001',
 '99',
 '97',
 '2001',
 '99',
 '1998',
 '99',
 '2001',
 '97',
 '97',
 '99',
 '98',
 '97',
 '1997',
 '1999',
 '1997',
 '97',
 '2000',
 '2001',
 '1999',
 '1999'

In [20]:
uniquebirthy = set(birthylist)

uniquebirthydict = {years:None for years in uniquebirthy}

for i in uniquebirthydict:
    yeartitle = input(i+": "+"Enter name: ")
    uniquebirthydict[i]=yeartitle

uniquebirthydict

99: Enter name: 1999
2001: Enter name: 2001
97: Enter name: 1997
1997: Enter name: 1997
1998: Enter name: 1998
1999: Enter name: 1999
98: Enter name: 1998
2000: Enter name: 2000


{'99': '1999',
 '2001': '2001',
 '97': '1997',
 '1997': '1997',
 '1998': '1998',
 '1999': '1999',
 '98': '1998',
 '2000': '2000'}

In [21]:
cleanbirthylist = [re.sub(year,uniquebirthydict[year],year) for year in birthylist]
cleanbirthylist

['1998',
 '2001',
 '1997',
 '1997',
 '1997',
 '1999',
 '1998',
 '1999',
 '1997',
 '2001',
 '1998',
 '1997',
 '1999',
 '1999',
 '1998',
 '1999',
 '1997',
 '1997',
 '1997',
 '1997',
 '1999',
 '1999',
 '1997',
 '1997',
 '1997',
 '1999',
 '1997',
 '2000',
 '1999',
 '1999',
 '1999',
 '1999',
 '1999',
 '1999',
 '2001',
 '1998',
 '2000',
 '2001',
 '1997',
 '1998',
 '1999',
 '1998',
 '1999',
 '1998',
 '1997',
 '1998',
 '1999',
 '1999',
 '1997',
 '2001',
 '1998',
 '1999',
 '2000',
 '1998',
 '1998',
 '1998',
 '2001',
 '1997',
 '1998',
 '1999',
 '1998',
 '2000',
 '2000',
 '1997',
 '1997',
 '1997',
 '1997',
 '2001',
 '1998',
 '1998',
 '1999',
 '1999',
 '1998',
 '1997',
 '1998',
 '1997',
 '2000',
 '2000',
 '2000',
 '1999',
 '1997',
 '1997',
 '2000',
 '1999',
 '1999',
 '1998',
 '1998',
 '1997',
 '1998',
 '1999',
 '1997',
 '1997',
 '1998',
 '1997',
 '2001',
 '1999',
 '1997',
 '1998',
 '1998',
 '1999',
 '1997',
 '2001',
 '2000',
 '2001',
 '1999',
 '1997',
 '2001',
 '1999',
 '1998',
 '1999',
 '2001',
 

In [22]:
cleanbirthytally={}

for i in cleanbirthylist:
    cleanbirthytally.setdefault(i, cleanbirthylist.count(i))
    
cleanbirthytally

{'1998': 746, '2001': 362, '1997': 756, '1999': 756, '2000': 380}

In [23]:
sortedyearlist = ["1997", "1998", "1999", "2000", "2001"]
sortedyeartally ={}

for i in sortedyearlist:
    sortedyeartally[i]= cleanbirthytally[i]
   
sortedyeartally

{'1997': 756, '1998': 746, '1999': 756, '2000': 380, '2001': 362}

In [24]:
#Checking to make sure all 3000 data was included
checkeryear = 0
for i in sortedyearlist:
    checkeryear += sortedyeartally[i]
    
checkeryear

3000

## Course List Tally

In [25]:
courselist= []

for i in range(1,3001):
    courselist.append(dirtyrecweek[i]["course"])
    
courselist

['BS LM',
 'Management',
 'BS CTM',
 'BS ITE',
 'BS CTM',
 'Management',
 'BS LM',
 'Management Engineering',
 'Management Engineering',
 'CTM',
 'BS LM',
 'BS ME',
 'BS CTM',
 'BS MGT',
 'Information Technology Entrepreneurship',
 'BS LM',
 'Information Technology Entrepreneurship',
 'BS MGT',
 'Information Technology Entrepreneurship',
 'BS ME',
 'LM',
 'ITE',
 'MGT',
 'ME',
 'BS ME',
 'BS ME',
 'CTM',
 'BS LM',
 'LM',
 'Communications Technology',
 'BS ITE',
 'Management',
 'BS ITE',
 'BS LM',
 'ITE',
 'BS MGT',
 'MGT',
 'Legal Management',
 'ME',
 'BS LM',
 'BS MGT',
 'BS LM',
 'Management',
 'Information Technology Entrepreneurship',
 'Management Engineering',
 'ME',
 'Management',
 'Information Technology Entrepreneurship',
 'Legal Management',
 'BS LM',
 'ITE',
 'Communications Technology',
 'BS MGT',
 'BS CTM',
 'Management Engineering',
 'ME',
 'BS CTM',
 'BS ME',
 'Information Technology Entrepreneurship',
 'ITE',
 'Communications Technology',
 'ITE',
 'BS ME',
 'Information 

In [33]:
uniquecourses = set(courselist)

uniquecoursedict = {years:None for years in uniquecourses}

for i in uniquecoursedict:
    yeartitle = input(i+": "+"Enter name: ")
    uniquecoursedict[i]=yeartitle

uniquecoursedict

BS ME: Enter name: BS Management Engineering
BS ITE: Enter name: BS IT Entrepreneurship
Information Technology Entrepreneurship: Enter name: BS IT Entrepreneurship
Communications Technology: Enter name: BS Communications Technology Management
ITE: Enter name: BS IT Entrepreneurship
LM: Enter name: BS Legal Management
Legal Management: Enter name: BS Legal Management
BS MGT: Enter name: BS Management
BS LM: Enter name: BS Legal Management
Management Engineering: Enter name: BS Management Engineering
CTM: Enter name: BS Communications Technology Management
MGT: Enter name: BS Management
BS CTM: Enter name: BS Communications Technology Management
ME: Enter name: BS Management Engineering
Management: Enter name: BS Management


{'BS ME': 'BS Management Engineering',
 'BS ITE': 'BS IT Entrepreneurship',
 'Information Technology Entrepreneurship': 'BS IT Entrepreneurship',
 'Communications Technology': 'BS Communications Technology Management',
 'ITE': 'BS IT Entrepreneurship',
 'LM': 'BS Legal Management',
 'Legal Management': 'BS Legal Management',
 'BS MGT': 'BS Management',
 'BS LM': 'BS Legal Management',
 'Management Engineering': 'BS Management Engineering',
 'CTM': 'BS Communications Technology Management',
 'MGT': 'BS Management',
 'BS CTM': 'BS Communications Technology Management',
 'ME': 'BS Management Engineering',
 'Management': 'BS Management'}

In [34]:
cleancourselist = [re.sub(course,uniquecoursedict[course],course) for course in courselist]
cleancourselist

['BS Legal Management',
 'BS Management',
 'BS Communications Technology Management',
 'BS IT Entrepreneurship',
 'BS Communications Technology Management',
 'BS Management',
 'BS Legal Management',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Communications Technology Management',
 'BS Legal Management',
 'BS Management Engineering',
 'BS Communications Technology Management',
 'BS Management',
 'BS IT Entrepreneurship',
 'BS Legal Management',
 'BS IT Entrepreneurship',
 'BS Management',
 'BS IT Entrepreneurship',
 'BS Management Engineering',
 'BS Legal Management',
 'BS IT Entrepreneurship',
 'BS Management',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Communications Technology Management',
 'BS Legal Management',
 'BS Legal Management',
 'BS Communications Technology Management',
 'BS IT Entrepreneurship',
 'BS Management',
 'BS IT Entrepreneurship',
 'BS Legal Management',
 'BS IT Entrepreneurship',
 'BS Manag

In [35]:
cleancoursetally={}

for i in cleancourselist:
    cleancoursetally.setdefault(i, cleancourselist.count(i))

cleancoursetally

{'BS Legal Management': 585,
 'BS Management': 629,
 'BS Communications Technology Management': 589,
 'BS IT Entrepreneurship': 590,
 'BS Management Engineering': 607}

In [36]:
#Check if all 3000 was included
listofcourses = {"BS Legal Management","BS Management","BS Communications Technology Management", "BS IT Entrepreneurship", "BS Management Engineering"}
checkeryear = 0
for i in listofcourses:
    checkeryear += cleancoursetally[i]
    
checkeryear

3000

## Cleaning Phone Numbers

In [37]:
phonenumberslist= []

for i in range(1,3001):
    phonenumberslist.append(dirtyrecweek[i]["phonenumber"])
    
phonenumberslist

['+63 918 302 6210',
 '927 212 6276',
 '+63 918 253 7017',
 '918 714 4360',
 '+63 927 462 1674',
 '+63 918 629 5268',
 '+63 917 369 7388',
 '917 240 6000',
 '918 226 5748',
 '+63 918 342 1717',
 '+63 927 708 1308',
 '927 747 5497',
 '+63 917 959 4490',
 '+63 927 543 4192',
 '917 211 9497',
 '+63 918 185 5162',
 '927 146 8146',
 '927 516 4280',
 '+63 927 958 6909',
 '+63 917 514 9105',
 '+63 927 984 7499',
 '+63 927 699 3694',
 '917 317 2478',
 '+63 917 159 3829',
 '+63 918 898 6972',
 '917 796 4622',
 '+63 927 138 7580',
 '+63 917 195 9244',
 '+63 917 643 3996',
 '917 758 3119',
 '+63 918 669 4265',
 '+63 917 750 8539',
 '927 236 9504',
 '917 438 4447',
 '+63 917 773 6035',
 '918 455 4185',
 '927 889 9252',
 '927 874 8083',
 '927 842 3053',
 '927 923 1234',
 '+63 918 381 4956',
 '+63 918 602 9278',
 '+63 918 532 6748',
 '917 654 1291',
 '918 244 1683',
 '917 818 1190',
 '+63 917 468 6425',
 '+63 918 635 1688',
 '917 821 1896',
 '927 924 9506',
 '918 854 2626',
 '+63 918 438 9745',
 '91

In [38]:
newphonenumberslist =[]
for i in range(0,3000):
    number = phonenumberslist[i]
    new_number = re.sub(r"^(9[0-9][0-9])",r"+63 \1",number)
    newphonenumberslist.append(new_number)
newphonenumberslist

['+63 918 302 6210',
 '+63 927 212 6276',
 '+63 918 253 7017',
 '+63 918 714 4360',
 '+63 927 462 1674',
 '+63 918 629 5268',
 '+63 917 369 7388',
 '+63 917 240 6000',
 '+63 918 226 5748',
 '+63 918 342 1717',
 '+63 927 708 1308',
 '+63 927 747 5497',
 '+63 917 959 4490',
 '+63 927 543 4192',
 '+63 917 211 9497',
 '+63 918 185 5162',
 '+63 927 146 8146',
 '+63 927 516 4280',
 '+63 927 958 6909',
 '+63 917 514 9105',
 '+63 927 984 7499',
 '+63 927 699 3694',
 '+63 917 317 2478',
 '+63 917 159 3829',
 '+63 918 898 6972',
 '+63 917 796 4622',
 '+63 927 138 7580',
 '+63 917 195 9244',
 '+63 917 643 3996',
 '+63 917 758 3119',
 '+63 918 669 4265',
 '+63 917 750 8539',
 '+63 927 236 9504',
 '+63 917 438 4447',
 '+63 917 773 6035',
 '+63 918 455 4185',
 '+63 927 889 9252',
 '+63 927 874 8083',
 '+63 927 842 3053',
 '+63 927 923 1234',
 '+63 918 381 4956',
 '+63 918 602 9278',
 '+63 918 532 6748',
 '+63 917 654 1291',
 '+63 918 244 1683',
 '+63 917 818 1190',
 '+63 917 468 6425',
 '+63 918 635

In [39]:
counterr = 0
unique= set(newphonenumberslist)
for number in unique:
    counterr += 1

counterr

3000

In [40]:
carierlist = []
cleancariertally={"Globe":0, "Smart":0}

for number in newphonenumberslist: #substitute numbers for 'Globe' and 'Smart'
    match = re.search(r'[+]63\s917\s[0-9]{3}\s[0-9]{4}', number)
    if match:
        cleancariertally["Globe"]+=1
    match2 = re.search(r'[+]63\s927\s[0-9]{3}\s[0-9]{4}', number)
    if match2:
        cleancariertally["Globe"]+=1
    match3 = re.search(r'[+]63\s918\s[0-9]{3}\s[0-9]{4}', number)
    if match3:
        cleancariertally["Smart"]+=1

cleancariertally

{'Globe': 2045, 'Smart': 955}

## Cleaning the list

In [51]:
i=1
j=1
k=1
l=1
o=1
cleanrecweek = dirtyrecweek.copy()
for month in cleanbirthmlist:
    cleanrecweek[i]["birthmonth"] = month
    i +=1

for day in birthdlist:
    cleanrecweek[j]["birthday"] = day
    j +=1

for year in cleanbirthylist:
    cleanrecweek[k]["birthyear"] = year
    k +=1

for course in cleancourselist:
    cleanrecweek[l]["course"] = course
    l +=1

for number in newphonenumberslist:
    dirtyrecweek[o]["phonenumber"] = number
    o +=1    
cleanrecweek

{1: {'firstname': 'name1',
  'lastname': 'lastname1',
  'birthmonth': 'November',
  'birthday': '26',
  'birthyear': '1998',
  'course': 'BS Legal Management',
  'phonenumber': '+63 918 302 6210'},
 2: {'firstname': 'name2',
  'lastname': 'lastname2',
  'birthmonth': 'May',
  'birthday': '26',
  'birthyear': '2001',
  'course': 'BS Management',
  'phonenumber': '+63 927 212 6276'},
 3: {'firstname': 'name3',
  'lastname': 'lastname3',
  'birthmonth': 'May',
  'birthday': '16',
  'birthyear': '1997',
  'course': 'BS Communications Technology Management',
  'phonenumber': '+63 918 253 7017'},
 4: {'firstname': 'name4',
  'lastname': 'lastname4',
  'birthmonth': 'December',
  'birthday': '9',
  'birthyear': '1997',
  'course': 'BS IT Entrepreneurship',
  'phonenumber': '+63 918 714 4360'},
 5: {'firstname': 'name5',
  'lastname': 'lastname5',
  'birthmonth': 'September',
  'birthday': '5',
  'birthyear': '1997',
  'course': 'BS Communications Technology Management',
  'phonenumber': '+63 

## Making cleanrecweekdata.csv

In [52]:
with open('cleanrecweekdata.csv', mode='w') as csv_file:
    headers = ['firstname', 'lastname', 'birthmonth', 'birthday', 'birthyear','course', 'phonenumber']
    writer = csv.DictWriter(csv_file, fieldnames=headers)
    writer.writeheader()
    for i in range(1,3000):
        writer.writerow(cleanrecweek[i])

In [54]:
cleancariertally

{'Globe': 2045, 'Smart': 955}

In [55]:
cleancoursetally

{'BS Legal Management': 585,
 'BS Management': 629,
 'BS Communications Technology Management': 589,
 'BS IT Entrepreneurship': 590,
 'BS Management Engineering': 607}

In [56]:
sortedyeartally

{'1997': 756, '1998': 746, '1999': 756, '2000': 380, '2001': 362}

In [57]:
sortedmonthtally

{'January': 270,
 'February': 232,
 'March': 246,
 'April': 267,
 'May': 273,
 'June': 259,
 'July': 249,
 'August': 241,
 'September': 232,
 'October': 178,
 'November': 277,
 'December': 276}

In [53]:
sorteddaytally

{1: 112,
 2: 106,
 3: 90,
 4: 110,
 5: 100,
 6: 99,
 7: 88,
 8: 111,
 9: 98,
 10: 111,
 11: 105,
 12: 98,
 13: 119,
 14: 102,
 15: 121,
 16: 115,
 17: 116,
 18: 107,
 19: 113,
 20: 91,
 21: 123,
 22: 103,
 23: 116,
 24: 108,
 25: 116,
 26: 116,
 27: 102,
 28: 104}

# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
         </table>
    </body>
</html>
```

In [58]:
#making the table for birthmonth
html_birthmonth = '''<table border="1">
<tr><th>Month</th><th>Count</th></tr>
'''
for i in sortedmonthtally:
    html_birthmonth += "<tr><td>{}</td><td>{}</td></tr>".format(i,sortedmonthtally[i])
html_birthmonth+= "</table>"

with open('birthm.html',"w+") as file:
    file.write(html_birthmonth)

In [59]:
#making the table for birthday
html_birthday = '''<table border="1">
<tr><th>Month</th><th>Count</th></tr>
'''
for i in sorteddaytally:
    html_birthday += "<tr><td>{}</td><td>{}</td></tr>".format(i,sorteddaytally[i])
html_birthday+= "</table>"

with open('birthd.html',"w+") as file:
    file.write(html_birthday)

In [60]:
#making the table for year
html_birthyear = '''<table border="1">
<tr><th>Month</th><th>Count</th></tr>
'''
for i in sortedyeartally:
    html_birthyear += "<tr><td>{}</td><td>{}</td></tr>".format(i,sortedyeartally[i])
html_birthyear+= "</table>"

with open('birthy.html',"w+") as file:
    file.write(html_birthyear)

In [100]:
#making the table for courses
html_courses = '''<table border="1">
<tr>'''
listofcourses= set(cleancourselist)
for cor in listofcourses: 
    html_courses += "<th>{}</th>".format(cor) #put the course names
html_courses+= "</tr>"
html_courses += "<tr><td>Count</td>"
for i in cleancoursetally:
    html_courses += "<td>{}</td>".format(cleancoursetally[i])
html_courses+= "</table>"

with open('courses.html',"w+") as file:
    file.write(html_courses)

In [101]:
#making the table for carriers
html_carrier = '''<table border="1">
<tr>'''
for i in cleancariertally:
    html_carrier += "<tr><td>{}</td><td>{}</td></tr>".format(i,cleancariertally[i])
html_carrier+= "</table>"

with open('carriers.html',"w+") as file:
    file.write(html_carrier)

In [102]:
htmlcode= '''
<html>
    <head>
    <style>
    #container {width: 800px;
    margin-left:auto;
    margin-right:auto;}
    h1   {background-color: #F67280;
    color:white}
    body {background-color: #F8B195;
     color:white}
    #births {padding: 10px}
    #month {background-color: #F67280;
    width: 300px;
    float: left;}
    #day {background-color: #F67280;
    width: 250px;
    float: left;}
    #year {background-color: #F67280;
    width: 250px;
    float: left;}
    #course {background-color: #F67280;}
    #carrier {background-color: #F67280;}
    </style>'''
#split into two because css wouldn't run when I had to format to add the other htmls
htmlpart2 = '''</head>
    <body>
         <div id="contain">
            <div id="header">
                <h1>Summary Statistics</h1>
            </div>
            <div id="births">
                <div id="month">
                {}
                </div>
                <div id="day">
                {}
                </div>
                <div id="year">
                {}
                </div>
            </div>
            <div id="course">
            {}
            </div>
            <div id="carrier">
            {}
            </div>
         </div>
    </body>
</html>'''
htmlpart2 = htmlpart2.format(html_birthmonth,html_birthday,html_birthyear,html_courses,html_carrier)
htmlcode += htmlpart2

In [103]:
with open('stats.html',"w+") as file:
    file.write(htmlcode)